src/test/java/com/hillel/items_exchange/exceptionshandling/GlobalControllerExceptionHandlerTest.java

```diff
@@ -0,0 +1,178 @@
+import com.hillel.items_exchange.controller.ImageController;
+import com.hillel.items_exchange.dto.AdvertisementDto;
+import com.hillel.items_exchange.exception.InvalidDtoException;
+import com.hillel.items_exchange.util.AdvertisementDtoCreatingUtil;
+import com.hillel.items_exchange.util.MessageSourceUtil;
+import org.hibernate.boot.model.naming.IllegalIdentifierException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import javax.persistence.EntityNotFoundException;
+import javax.validation.ConstraintViolationException;
+
+import static com.hillel.items_exchange.util.JsonConverter.asJsonString;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class GlobalControllerExceptionHandlerTest {
+
+    @Autowired
+    private WebApplicationContext context;
+
+    private MockMvc mockMvc;
+    private AdvertisementDto nonExistDto;
+    private AdvertisementDto existDto;
+
+    @Mock
+    AdvertisementController advertisementController;
+
+    @Mock
+    CategoryController categoryController;
+
+    @Mock
+    ImageController imageController;
+
+    @Mock
+    MessageSourceUtil messageSourceUtil;
+
+    @BeforeEach
+    void setup() {
+        nonExistDto = AdvertisementDtoCreatingUtil.createNonExistAdvertisementDto();
+        existDto = AdvertisementDtoCreatingUtil.createExistAdvertisementDto();
+        mockMvc = MockMvcBuilders.standaloneSetup(advertisementController, categoryController)
+                .setControllerAdvice(new GlobalControllerExceptionHandler(messageSourceUtil))
+                .build();
+    }
+
+    @Test
+    public void testHandleUserNotFoundException() throws Exception {
+        when(advertisementController.createAdvertisement(any(), any())).thenThrow(UsernameNotFoundException.class);
+
+        MvcResult result = mockMvc.perform(post("/adv")
```

Comment:
This part could be refactored as well (put, post):
`getResult(MockHttpServletRequestBuilder requestType, AdvertisementDto dto, ResultMatcher result) throws Exception`

Reply:
I did a little different and now we can use `getResult` method not only for put and post http methods